lib/url.js

```diff
@@ -51,7 +51,9 @@ const nonHostChars = ['%', '/', '?', ';', '#'].concat(autoEscape);
 const hostEndingChars = ['/', '?', '#'];
 const hostnameMaxLen = 255;
 const hostnamePartPattern = /^[+a-z0-9A-Z_-]{0,63}$/;
-const hostnamePartStart = /^([+a-z0-9A-Z_-]{0,63})(.*)$/;
+const hostnamePartStart = /^([^:\/?#\t\r\n]{0,63})(.*)$/;
+const hostnameTerminators = /[:\/?#\t\r\n]/;
+const hostnameEncode = /[^+a-z0-9A-Z_\-.]/g;
```

Comment:
I tend to favor `\w` and other character classes but in this case, I was mimicking the pre-existing regexp literal on line 53.